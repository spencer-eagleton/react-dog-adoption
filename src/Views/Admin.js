import AdminForm from '../Components/AdminForm';

export default function Admin() {
  return <AdminForm />;
}
