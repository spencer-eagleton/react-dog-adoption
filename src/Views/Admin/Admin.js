import AdminForm from '../../Components/AdminForm';
import { useState } from 'react';
import { createDog } from '../../services/doglist';
import { useHistory } from 'react-router-dom';

export default function Admin() {
  const [name, setName] = useState('');
  const [breed, setBreed] = useState('');
  const [age, setAge] = useState('');
  const [image, setImage] = useState('');
  const [bio, setBio] = useState('');
  const [updatePrompt, setUpdatePrompt] = useState('');
  const history = useHistory();

  const submit = async (e) => {
    e.preventDefault();

    await createDog(name, breed, age, image, bio);
    setUpdatePrompt('Doggo added!');
    setTimeout(() => {
      history.push('/');
    }, 3000);
  };

  return (
    <div>
      <p>{updatePrompt}</p>
      <AdminForm
        name={name}
        setName={setName}
        breed={breed}
        setBreed={setBreed}
        age={age}
        setAge={setAge}
        image={image}
        setImage={setImage}
        bio={bio}
        setBio={setBio}
        handleSubmit={submit}
      />
    </div>
  );
}
