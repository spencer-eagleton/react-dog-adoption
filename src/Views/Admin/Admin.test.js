import Admin from './Admin';
import { render } from '@testing-library/react';
test('checking for admin component', async () => {
  const { container } = await render(<Admin />);
  expect(container).toMatchSnapshot();
});
