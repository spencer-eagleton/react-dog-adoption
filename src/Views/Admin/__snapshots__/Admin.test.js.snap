// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking for admin component 1`] = `
<div>
  <div>
    <p />
    <div>
      Add a new pupperoo
    </div>
    <form>
      <label>
        Name:
        <input
          placeholder="Enter Name"
          type="text"
          value=""
        />
      </label>
      <label>
        Breed:
        <input
          placeholder="Enter Breed"
          type="text"
          value=""
        />
      </label>
      <label>
        Age:
        <input
          placeholder="Enter Age"
          type="number"
          value=""
        />
      </label>
      <label>
        Image:
        <input
          placeholder="Image URL"
          type="text"
          value=""
        />
      </label>
      <label>
        Bio:
        <textarea
          placeholder="Enter Bio"
        />
      </label>
      <button>
        Update Puppers
      </button>
    </form>
  </div>
</div>
`;
